ffi/src/lib.rs

```diff
@@ -388,14 +402,98 @@ fn create_client_internal(
-    Ok(ClientAdapter { runtime, core })
+    // Clone client_adapter before moving it into the async block
+    let client_adapter_clone = Arc::into_raw(client_adapter.clone()).addr();
+
+    // Check if the pubsub_callback function pointer is not null
+    // We can't directly use is_null() on function pointers, so we need to compare with std::ptr::null()
+    let null_fn: PubSubCallback = unsafe { std::mem::transmute(std::ptr::null::<()>()) };
+    let is_callback_valid = !std::ptr::eq(
+        pubsub_callback as *const () as *const u8,
+        null_fn as *const () as *const u8,
+    );
+
+    // If pubsub_callback is provided (not null), spawn a task to handle push notifications
+    if is_callback_valid {
+        runtime.spawn(async move {
+            loop {
+                let result = push_rx.recv().await;
+                match result {
+                    None => {
+                        return;
+                    }
+                    Some(push_msg) => {
+                        // Convert the push_msg.data to a CommandResponse
+                        let data_response = match valkey_value_to_command_response(Value::Array(
+                            push_msg.data.clone(),
+                        )) {
+                            Ok(response) => response,
+                            Err(_e) => {
+                                continue; // Skip this message if conversion fails
+                            }
+                        };
+
+                        // Get the numeric value of the PushKind enum
+                        let kind_value = match push_msg.kind {
+                            redis::PushKind::Disconnection => 0,
+                            redis::PushKind::Other(_) => 1,
+                            redis::PushKind::Invalidate => 2,
+                            redis::PushKind::Message => 3,
+                            redis::PushKind::PMessage => 4,
+                            redis::PushKind::SMessage => 5,
+                            redis::PushKind::Unsubscribe => 6,
+                            redis::PushKind::PUnsubscribe => 7,
+                            redis::PushKind::SUnsubscribe => 8,
+                            redis::PushKind::Subscribe => 9,
+                            redis::PushKind::PSubscribe => 10,
+                            redis::PushKind::SSubscribe => 11,
+                        };
+
+                        // Call the pubsub callback with the push notification data
+                        unsafe {
+                            pubsub_callback(
+                                client_adapter_clone,
+                                kind_value,
+                                Box::into_raw(Box::new(data_response)),
+                            );
+                        }
+                    }
+                }
+            }
+        });
+    } else {
+        // If no pubsub callback is provided, we need to handle the channel to prevent infinite queue growth
```

Comment:
If no callback provided, we should give no tx on line 412

Reply:
Not sure what you are saying here. Are you saying that the system will never send a message on push_tx if there is no callback defined and therefore we don't need to check it again here?

Reply:
Yes, exactly. If we don't provide a `tx`, we have no reason to listen on `rx`.

Reply:
Merged your fix.